controller/array_action/array_mediator_interface.py

```diff
@@ -131,6 +131,23 @@ def get_volume_name(self, volume_id):
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def get_volume_by_id(self, volume_id):
+        """
+            This function return volume info about the volume.
+
+            Args:
+                volume_id: id of the volume on storage system.
+
+
+            Returns:
+               Volume
+
+            Raises:
+                VolumeNotFound
```

Comment:
in other places (e.g. snapshots), we [return None](https://github.com/IBM/ibm-block-csi-driver/pull/177#discussion_r440113809).
do you think the behavior should be aligned? and if so, to which of the approaches?

Reply:
in [get_volume, we return NotFound](https://github.com/IBM/ibm-block-csi-driver/blob/cc133625139db6729e4b2dbd582eb8de2c70eb38/controller/array_action/array_mediator_interface.py#L113).

Reply:
IMO `get_volume` should also return `None`, but I'm not sure regarding `get_volume_by_id`.
what do you think?

Reply:
Idk, can you share your reasons?

Reply:
for `get_volume`, changing to `None` will save `CreateVolume` a nested `try`.
for `get_volume_by_id`, changing to `None` will be aligned to the `get_volume` change, but will complicate `ValidateVolumeCapabilities`

Reply:
ok, so you are suggesting to change get_volume.. its not really in this ticket scope

Reply:
ok, already in CSI-1446